### ISSUE TITLE
Replace CraftOS pager with a better one

### ISSUE THREAD

**Issue by apemanzilla** [SquidDev-CC/mbs#16]
`textutils.pagedPrint` and `textutils.pagedTabulate` are pretty bad, replacing them with a custom pager that can be scrolled properly (basically `less`) would be nice.

**Comment by SquidDev** [comment]
I'm not entirely sure how this will function with `help`. This prints each line in the file with a separate call to `textutils.pagedPrint`, which makes it much harder to have one unified pager.

**Comment by apemanzilla** [comment]
In that case, why not just write a custom `help` program and set the path so that it's resolved first?

**Comment by SquidDev** [comment]
Yeah, I probably will. Might also PR that change into CC - there's a couple of other minor tweaks I'd like to make to the Lua side of things.

The other thing worth bearing in mind is whether the pager behaves like `less` or `less -F`, and if there's any situations we might want to disable it entirely. For instance, I'm not sure if `list` (`ls`/`dir`) should use a pager or just print everything to the terminal and allow people to scroll.

I guess I could provide some setting which allows people to control that (as well as program-specific control or something), but it seems a little cumbersome.

**Comment by apemanzilla** [comment]
You could also replace the `list` program if necessary :stuck_out_tongue:

**Comment by Lupus590** [comment]
Related?
https://github.com/SquidDev-CC/CC-Tweaked/pull/602
https://github.com/SquidDev-CC/CC-Tweaked/pull/603

**Comment by SquidDev** [comment]
@Lupus590 Yep. Currently neither change really touches `textutils.pagedTabulate`, which is the other big function - I suspect that'll need to wait until I've reviewed SquidDev-CC/CC-Tweaked#522.

Speaking of which, I haven't forgotten your PR, just laptop is currently dead, so there's a limit of how much work I can do.
